### PR TITLE
Bunch of minor fixes

### LINKED ISSUE #458: Log-file bug in md-flexible
**Describe the bug**
When a log-file is given in the user input, the output is still in the console and the file is empty.

**To Reproduce**
1. Build md-flex
2. run md-flex with --log-file anyFileName --log-level debug
3. See that output is in console and file is empty

**Expected behavior**
The output of the logger should be in the given filename.

**Possible explentation**
The file name is never given to the AutoPas object and therefore not output can be written in it.
The file name needs to be given with the initiation of the AutoPas object or it has to be changed after the initiation.

### PR DESCRIPTION
**Pull request by FG-TUM**
# Description

several small stuff I stumbled upon while reviewing code.
- [x] added lots of `[[nodiscard]]` in the traversals
- [x] merge namespaces for `OptimumSelector`
- [x] c04: convert computeOffsets32Pack() to a stateless function and use it in the initializer list of the constructor
- [x] `AutoPas.h` move default values from constructor to declarations.
- [x] Remove `AUTOPAS_WARN_UNUSED`
- [x] fix #458 
- [x] when invoking `md-flexible` with `--help` sort options

# How Has This Been Tested?

just existing tests.

### CODE REVIEW

File: examples/md-flexible/parsing/CLIParser.cpp [hunk 1]
```diff
@@ -494,13 +494,22 @@ bool CLIParser::parseInput(int argc, char **argv, MDFlexConfig &config) {
...
+    std::vector<std::pair<std::string, bool>> options;
+    for (auto &o : long_options) {
+      if (o.name != nullptr) {
+        options.emplace_back(std::make_pair(o.name, o.has_arg));
       }
-      cout << "    --" << setw(MDFlexConfig::valueOffset + 2) << left << o.name;
-      if (o.has_arg) {
+    }
+
+    // by default sort sorts by first member of pair
```
Comment (SteffenSeckler) [inline examples/md-flexible/parsing/CLIParser.cpp:505]:
```suggestion
    // By default `std::sort` sorts by first member of pair.
```

File: examples/md-flexible/parsing/CLIParser.cpp [hunk 2]
```diff
@@ -526,7 +535,7 @@ bool checkFileExists(const std::string &filename) {
 }  // namespace
 
 void CLIParser::inputFilesPresent(int argc, char **argv, MDFlexConfig &config) {
-  int option, optionIndex;
+  int option = 0, optionIndex = 0;
```
Comment (SteffenSeckler) [inline examples/md-flexible/parsing/CLIParser.cpp:538]: You might want to make the while loop below into a for loop and move the current `option` and `optionIndex` into its initializer statement.
You might also want to rename the variable `option`, as currently it just creates confusion between the class `option` and the variable `option`.

Comment (FG-TUM) [reply examples/md-flexible/parsing/CLIParser.cpp:538]: And how do you propose this for loop header should look like?
```C++
  for (int cliOptionIndex = 0, cliOption = cliOption = getopt_long(argc, argv, "", long_options, &cliOptionIndex);
       cliOption != -1; cliOption = cliOption = getopt_long(argc, argv, "", long_options, &cliOptionIndex)) {
```
this does not look very appealing and just creates code duplication and more confusion :/

Comment (SteffenSeckler) [reply examples/md-flexible/parsing/CLIParser.cpp:538]:
```c++
for (int option = 0, optionIndex = 0; (option = getopt_long(argc, argv, "", longOptions, &optionIndex)) != -1; ) {
```

Comment (SteffenSeckler) [reply examples/md-flexible/parsing/CLIParser.cpp:538]: just to have things in the proper scope.

Comment (FG-TUM) [reply examples/md-flexible/parsing/CLIParser.cpp]: ugly but I like it.

Comment (FG-TUM) [reply examples/md-flexible/parsing/CLIParser.cpp]: done in 7825fc3

**Pull request merged**